src/package-request.js

```diff
@@ -281,7 +281,8 @@ export default class PackageRequest {
       promises.push(this.resolver.find({
         pattern: depPattern,
         registry: remote.registry,
-        optional: false,
+        // dependencies of optional dependencies should themselves be optional
+        optional: this.optional,
```

Comment:
Just a gut feeling, we had problems with deeper chains of dependency :)
Probably 2 layers should be good enough